### PR TITLE
[teleop] New Teleop/Auton Mode interface and teleop mode button on auton GUI

### PR DESCRIPTION
**Pull request by jnnanni**
-Added Teleop Mode button to auton GUI
-updated nav box color logic 
-added Enable.lcm and new interface for auton/teleop enabled

image

### CODE REVIEW

File: base_station/gui/src/components/LCMEcho.vue [hunk 1]
```diff
@@ -175,6 +176,7 @@
           {'topic': '/target_bearing', 'type': 'TargetBearing'},
           {'topic': '/target_list', 'type': 'TargetList'},
           {'topic': '/target_orientation', 'type': 'TargetOrientation'},
+          {'topic': '/teleop_en', 'type': 'Enable'},
```
Comment (cgiger00) [inline base_station/gui/src/components/LCMEcho.vue:179]: should ```{'topic': '/auton', 'type': 'AutonState'}``` be auton_en and Enable?

**Review by cgiger00 — CHANGES_REQUESTED** [review]

File: base_station/gui/src/components/LCMEcho.vue (hunk 1 above)
Comment (CameronTressler) [reply base_station/gui/src/components/LCMEcho.vue:179]: This^ but also, can we just write out /auton_enabled and /teleop_enabled?

**Comment by jnnanni** [comment]
Newly added /auton_led lcm channel to tell ESW what color to change the auton LED to.

**Review by tabiosg — CHANGES_REQUESTED** [review]
wait until ESW updates their end of stuff IMO, i'll make a PR by tn

**Comment by tabiosg** [comment]
plz accept this https://github.com/jnnanni/mrover-workspace/pull/2

**Review by tabiosg — APPROVED** [review]

File: base_station/gui/src/components/LCMEcho.vue (hunk 1 above)
Comment (jnnanni) [reply base_station/gui/src/components/LCMEcho.vue:179]: My bad I didn't see this until now, will make changes later tonight or tomorrow morning

File: jetson/science_bridge/src/__main__.py [hunk 2]
```diff
@@ -61,13 +60,12 @@ def __init__(self):
...
             "TRIAD": self.triad_handler,
             "TXT": self.txt_handler,
-            "CAROUSEL": self.carousel_handler,
             "HEATER": self.heater_state_handler,
             "AUTOSHUTOFF": self.heater_shutoff_handler
         }
-        self.last_openloop_cmd = -1
         self.max_error_count = 20
         self.sleep = .01
+        self.previous_led_state = LED_state.NONE
```
Comment (cgiger00) [inline jetson/science_bridge/src/__main__.py:68]: again with the camal case snake case mix...

File: jetson/science_bridge/src/__main__.py [hunk 3]
```diff
@@ -201,156 +203,71 @@ def heater_transmit(self, channel, msg):
...
-            green_off = message.format(device=Mosfet_devices.GREEN_LED.value, enable=0)
-            green_off = self.add_padding(green_off)
-            self.ser.write(bytes(green_off, encoding='utf-8'))
-        if (prev != Auton_state.ELSE):
-            red_off = message.format(device=Mosfet_devices.RED_LED.value, enable=0)
-            red_off = self.add_padding(red_off)
-            self.ser.write(bytes(red_off, encoding='utf-8'))
+    def auton_led(self, channel, msg):  # TODO - fix make better
+        # Off, Done, On
+        struct = AutonLed.decode(msg)
```
Comment (cgiger00) [inline jetson/science_bridge/src/__main__.py:230]: rename to auton_led instead of struct for readability?

File: jetson/science_bridge/src/__main__.py [hunk 4]
```diff
@@ -201,156 +203,71 @@ def heater_transmit(self, channel, msg):
...
+            requested_state = LED_state.GREEN
+        elif struct.color == "Blue":
+            requested_state = LED_state.BLUE
+
+        if self.previous_led_state == requested_state:
+            return
+
+        self.previous_led_state = requested_state
+        if requested_state == LED_state.RED:
+            print("Received new auton led request: Turning RED")
```
Comment (cgiger00) [inline jetson/science_bridge/src/__main__.py:246]: can these be part of a debug thing? i would rather not have these merged into main as always printing during led state changes

File: base_station/gui/src/components/AutonTask.vue [hunk 5]
```diff
@@ -168,37 +168,58 @@ export default {
...
     }),
-  },
-
-  created: function () {
 
-    setInterval(() => {
-      if(this.nav_status.nav_state_name == "Off"){
-        this.nav_state_color = navBlue
+    nav_state_color: function() {
+      if(this.teleopEnabled){
```
Comment (cgiger00) [inline base_station/gui/src/components/AutonTask.vue:176]: is the teleop enabled/auton led thing added to the gui read me?

File: jetson/science_bridge/src/__main__.py [hunk 6]
```diff
@@ -201,156 +203,71 @@ def heater_transmit(self, channel, msg):
...
+        requested_state = LED_state.NONE
+
+        if struct.color == "Red":
+            requested_state = LED_state.RED
+        elif struct.color == "Green":
+            requested_state = LED_state.GREEN
+        elif struct.color == "Blue":
+            requested_state = LED_state.BLUE
+
+        if self.previous_led_state == requested_state:
```
Comment (cgiger00) [inline jetson/science_bridge/src/__main__.py:241]: is teleop constantly spamming the nav state changes? i feel like it'll be at lot cleaner for this state change logic to be handled on their part since it's already done for the gui?

**Review by cgiger00 — CHANGES_REQUESTED** [review]

File: jetson/science_bridge/src/__main__.py (hunk 2 above)
Comment (tabiosg) [reply jetson/science_bridge/src/__main__.py:68]: changing to LEDState

File: jetson/science_bridge/src/__main__.py (hunk 3 above)
Comment (tabiosg) [reply jetson/science_bridge/src/__main__.py:230]: sure

File: jetson/science_bridge/src/__main__.py (hunk 6 above)
Comment (tabiosg) [reply jetson/science_bridge/src/__main__.py:241]: i'm not sure if teleop constantly spams nav state changes, @jnnanni do you?

File: jetson/science_bridge/src/__main__.py (hunk 4 above)
Comment (tabiosg) [reply jetson/science_bridge/src/__main__.py:246]: ok

File: jetson/science_bridge/src/__main__.py (hunk 6 above)
Comment (cgiger00) [reply jetson/science_bridge/src/__main__.py:241]: he said it doesn't, does the esw code handle that properly then?

Comment (jnnanni) [reply jetson/science_bridge/src/__main__.py:241]: I talked to Guthrie directly about it on slack, should be fine.

File: rover_msgs/AutonState.lcm [hunk 7]
```diff
@@ -1,5 +0,0 @@
-package rover_msgs;
```
Comment (tabiosg) [inline rover_msgs/AutonState.lcm:1]: why is this deleted?? jetson-teleop crashes now since auton state doesn't exist. is this file not supposed to be deleted or is jetson/teleop supposed to be changed?

**Review by tabiosg — CHANGES_REQUESTED** [review]
see comments about autonstate, fix asap!

**Comment by tabiosg** [comment]
I would probably close this PR since @ankithu opened up #1094
